mteb/abstasks/AbsTaskClusteringFast.py

```diff
@@ -133,17 +149,25 @@ def _evaluate_monolingual(
         logger.info(f"Encoding {len(downsampled_dataset)} sentences...")
 
         embeddings = model.encode(downsampled_dataset["sentences"])
+        labels = []
+        for label in downsampled_dataset["labels"]:
+            if not isinstance(label, list):
+                label = [label]
+            labels.append(label)
 
         v_measures = evaluate_clustering_bootstrapped(
             embeddings,
             downsampled_dataset["labels"],
             n_clusters=self.n_clusters,
             cluster_size=self.max_documents_per_cluster,
             kmean_batch_size=self.k_mean_batch_size,
+            max_depth=self.max_depth,
```

Comment:
max-depth in the categories list I suppose? I would probably move that to the creation of the list for the specific dataset. Or is there good reason to now have it?

Reply:
I mean could have it either way, but this limits it at a reasonable level for all datasets. Also specific tasks wouldn't have to write `dataset_transform()` methods to limit the number of levels, but could simply specify `max_depth`.

```python
class SomeSpecificTaskClustering(AbsTaskClusteringFast):
     max_depth = 3
```

We can also make the default `None` and include all levels, but I still believe it's a helpful attribute to have.